src/lib/hpke/DY.Lib.HPKE.Spec.fst

```diff
@@ -0,0 +1,67 @@
+module DY.Lib.HPKE.Spec
+
+open Comparse
+open DY.Core
+open DY.Lib.Comparse.Glue
+open DY.Lib.Comparse.Parsers
+
+#set-options "--fuel 0 --ifuel 1"
+
+/// This module defines the specification of HPKE.
+/// More precisely, this specification corresponds to the single-shot API in base mode
+/// (c.f. section 6.1 of RFC 9180 [1]).
+///
+/// This specification has the following limitations:
+/// - the message formatting is not compliant with RFC 9180 (e.g. the one in LabeledExpand)
+/// - some KDF.Extract were omitted, for example:
+///   - `info` is extracted before going to LabeledExpand's info
+///   - `shared_secret` is extracted with the empty PSK
+///      before going to LabeledExpand's prk
+///   (c.f. section 5.1 of RFC 9180 [2])
+/// Having a more compliant specification is left as future work.
+///
+/// [1] https://www.rfc-editor.org/rfc/rfc9180.html#name-encryption-and-decryption-2
+/// [2] https://www.rfc-editor.org/rfc/rfc9180.html#name-creating-the-encryption-con
+
+/// Definition of the internal function LabeledExpand
+
+[@@ with_bytes bytes]
+type hpke_labeled_expand_info = {
+  len: nat_lbytes 2;
+  [@@@ with_parser #bytes ps_string]
+  label: string;
```

Comment:
I agree that in general we should avoid naming things `label` as it is confusing with security labels, here I think we should keep the name because it is [the one used in the HPKE RFC](https://www.rfc-editor.org/rfc/rfc9180.html#section-4-10)

Reply:
That makes sense, but can you write this into a comment so that it is not confusing?

Reply:
Done